pkg/schedule/schedule.go

```diff
@@ -56,14 +58,20 @@ type fifo struct {
 
 	finishCond *sync.Cond
 	donec      chan struct{}
+	lg         *zap.Logger
 }
 
 // NewFIFOScheduler returns a Scheduler that schedules jobs in FIFO
 // order sequentially
-func NewFIFOScheduler() Scheduler {
+func NewFIFOScheduler(lg *zap.Logger) Scheduler {
+	if lg == nil {
```

Comment:
We decided to `verify.Assert` (client/pkg/verify/verify.go) whether the logger is prided rather than silently turn off logging if it was not provided.